core/src/main/java/net/sourceforge/jnlp/runtime/classloader/DelegatingClassLoader.java

```diff
@@ -0,0 +1,59 @@
+package net.sourceforge.jnlp.runtime.classloader;
+
+import java.net.URL;
+import java.util.Objects;
+
+public class DelegatingClassLoader extends ClassLoader {
+    public static final DelegatingClassLoader instance = new DelegatingClassLoader(Thread.currentThread().getContextClassLoader());
+    private ClassLoader classLoader;
+
+    public static DelegatingClassLoader getInstance() {
+        return instance;
+    }
+
+    DelegatingClassLoader(ClassLoader loader) {
+        super(loader);
+        this.classLoader = loader;
+    }
+
+    public void setClassLoader(ClassLoader loader) {
+        this.classLoader = loader;
```

Comment:
Null Check is needed

Reply:
fixed

Reply:
more easy: `this.classLoader = Assert.requireNonNull(loader, "loader");`

Reply:
fixed